sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -558,7 +574,23 @@ long flush(
               appendFailures.inc();
               return RetryType.RETRY_ALL_OPERATIONS;
             },
-            c -> recordsAppended.inc(c.protoRows.getSerializedRowsCount()),
+            c -> {
+              recordsAppended.inc(c.protoRows.getSerializedRowsCount());
+              if (successfulRowsReceiver != null) {
+                for (ByteString rowBytes : c.protoRows.getSerializedRowsList()) {
+                  try {
+                    TableRow row =
+                        TableRowToStorageApiProto.tableRowFromMessage(
+                            DynamicMessage.parseFrom(
+                                Preconditions.checkStateNotNull(appendClientInfo).getDescriptor(),
+                                rowBytes));
+                    successfulRowsReceiver.output(row);
```

Comment:
This assumes that there are no re-windowing of the data into the Global Window in the previous steps. Is it correct?

Reply:
I'm not sure what you mean. I think the data is in toe Global Window by now.

Reply:
OK, makes sense - data is converted to the global window as part of conversion into the proto message.